### ISSUE TITLE
Add spatial convergent cross mapping

### ISSUE THREAD

**Issue by rafaqz** [JuliaDynamics/CausalityTools.jl#368]
Would be nice to have a spatial ccm implementation here, especially for ecological applications.

See:
https://cran.r-project.org/web/packages/multispatialCCM/multispatialCCM.pdf
https://esajournals.onlinelibrary.wiley.com/doi/full/10.1890/14-1479.1

**Comment by rafaqz** [comment]
Probably the dew-drop regression component could be reused for other methods like PAI and CCS as well

**Comment by kahaaga** [comment]
Yep, I think by some careful design here, we can re-use the dewdrop regression for the other methods. That would be brilliant. 

Do you have any rough sketches of an implementation ready, or is it just it-would-be-nice-to-have at the moment?

**Comment by rafaqz** [comment]
Nothing specific yet. I'm not enough across how things work here to know how to fit new algs into the existing interface.
